end_to_end_tests/scenario_usage_tracking.py

```diff
@@ -0,0 +1,146 @@
+from datetime import datetime, timedelta, UTC
+import logging
+
+import asciichartpy as asciichart
+from azure.identity import DefaultAzureCredential
+from locust import HttpUser, task, constant, events
+
+import random
+
+from common.log_analytics import (
+    GroupDefinition,
+    QueryProcessor,
+)
+from common.latency import (
+    set_simulator_completions_latency,
+    report_request_metric,
+)
+from common.config import (
+    apim_keys,
+    simulator_endpoint_payg1,
+    simulator_endpoint_payg2,
+    tenant_id,
+    subscription_id,
+    resource_group_name,
+    app_insights_connection_string,
+    log_analytics_workspace_id,
+    log_analytics_workspace_name,
+)
+
+test_start_time = None
+deployment_name = "gpt-35-turbo-100m-token"
+
+
+class CompletionUser(HttpUser):
+    """
+    CompletionUser makes calls to the OpenAI Completions endpoint to show traffic via APIM
+    """
+
+    wait_time = constant(1)  # wait 1 second between requests
+
+    @task
+    def get_completion(self):
+        url = f"openai/deployments/{deployment_name}/completions?api-version=2023-05-15"
+        payload = {
+            "model": "gpt-5-turbo-1",
+            "prompt": "Once upon a time",
+            "max_tokens": 10,
+        }
+        apim_key = get_random_key(apim_keys)
```

Comment:
Great idea! Right now, the 3 subscriptions consume essentially the same number of tokens during the test. Using different max_tokens values would allow for differences in total tokens consumed.

Reply:
I elected to keep a single task but randomize the subscription key used and max tokens. I considered doing a task for each subscription key, but this solution was the least code/most readable, in my opinion.